homeassistant/components/withings/binary_sensor.py

```diff
@@ -22,9 +26,24 @@ async def async_setup_entry(
     """Set up the sensor config entry."""
     coordinator = hass.data[DOMAIN][entry.entry_id].bed_presence_coordinator
 
-    entities = [WithingsBinarySensor(coordinator)]
+    ent_reg = er.async_get(hass)
+
+    callback: Callable[[], None] | None = None
+
+    def _async_add_bed_presence_entity() -> None:
+        """Add bed presence entity."""
+        entities = [WithingsBinarySensor(coordinator)]
+
+        async_add_entities(entities)
+        if callback:
+            callback()
 
-    async_add_entities(entities)
+    if ent_reg.async_get_entity_id(
+        Platform.BINARY_SENSOR, DOMAIN, f"withings_{entry.unique_id}_in_bed"
+    ):
+        _async_add_bed_presence_entity()
+    else:
+        callback = coordinator.async_add_listener(_async_add_bed_presence_entity)
```

Comment:
You have this pattern in a few other places, it might be possible to make it a function

Reply:
I'll look into this for a follow up